chia/simulator/full_node_simulator.py

```diff
@@ -144,3 +173,219 @@ async def reorg_from_index_to_new_index(self, request: ReorgProtocol):
 
         for block in more_blocks:
             await self.full_node.respond_block(RespondBlock(block))
+
+    async def process_blocks(self, count: int, farm_to: bytes32 = bytes32([0] * 32)) -> int:
+        """Process the requested number of blocks including farming to the passed puzzle
+        hash. Note that the rewards for the last block will not have been processed.
+        Consider `.farm_blocks()` or `.farm_rewards()` if the goal is to receive XCH at
+        an address.
+
+        Arguments:
+            count: The number of blocks to process.
+            farm_to: The puzzle hash to farm the block rewards to.
+
+        Returns:
+            The total number of reward mojos for the processed blocks.
+        """
+        rewards = 0
+        height = uint32(0)
+
+        if count == 0:
+            return rewards
+
+        for _ in range(count):
+            block: FullBlock = await self.farm_new_transaction_block(FarmNewBlockProtocol(farm_to))
+            height = uint32(block.height)
+            rewards += calculate_pool_reward(height) + calculate_base_farmer_reward(height)
+
+        while True:
```

Comment:
I coded this so long ago that I don't recall if I got here by pure speculation or because I hit some funny case and needed it.  `:[`  If I can't find a reason for it, it can go away.